plugins/modules/iam_user.py

```diff
@@ -262,9 +263,20 @@ def create_or_update_login_profile(connection, module):
 
     try:
         retval = connection.update_login_profile(**user_params)
+        if wait:
+            AWSRetry.jittered_backoff(
+                delay=1,
+                catch_extra_error_codes=['EntityTemporarilyUnmodifiable']
+            )(connection.update_login_profile)(**user_params)
     except is_boto3_error_code('NoSuchEntity'):
+        # Login profile does not yet exist - create it
         try:
             retval = connection.create_login_profile(**user_params)
+            if wait:
+                AWSRetry.jittered_backoff(
+                    delay=1,
+                    catch_extra_error_codes=['EntityTemporarilyUnmodifiable']
```

Comment:
I like this approach, however looking at what this is doing it's probably worth applying this backoff (including the extra error_code logic) down to the initial connection creation.  Then you can use `aws_retry=True` for pretty much all of the calls, and also handle rate-limiting issues cleanly.

Reply:
The _only_ reason I didn't do that is because in that case, we're pretty much forcing the user to wait

Reply:
In my opinion, waiting to avoid something breaking is reasonable.  For most folks the 'wait' delay they don't expect is waiting for the change they made to finish propagating:

- An `EntityTemporarilyUnmodifiable` exception thrown by the initial update is something we want to retry on, and often do.
- The second call to update, with it's `EntityTemporarilyUnmodifiable` exception handling as a proxy for "something's still being changed", would be what folks wouldn't expect on `wait=False`.

Reply:
In theory you could use max_delay == delay but the jitter's going to mess it up a little.